test/e2e/page-objects/pages/test-dapp.ts

```diff
@@ -1140,6 +1171,92 @@ class TestDapp {
     console.log(`Check testdapp is connected to network ${networkId}`);
     await this.driver.waitForSelector(this.connectedNetwork(networkId));
   }
+
+  /**
+   * Check the number of dapps connected
+   *
+   * @param numberOfDapps - The number of dapps connected
+   */
+  async checkNumberOfDappsConnected(numberOfDapps: string) {
+    await this.driver.waitForSelector(this.dappNumberConnected(numberOfDapps));
+  }
+
+  /**
+   * Click the cancel button
+   */
+  async clickCancelButton() {
+    await this.driver.clickElement(this.dappCancelButton);
+  }
+
+  /**
+   * Click the confirm button
+   */
+  async clickConfirmButton() {
+    await this.driver.clickElement(this.dappConfirmButton);
+  }
+
+  /**
+   * Click the footer next button in a MetaMask dialog and wait for window to close
+   */
+  async clickFooterNextButton(): Promise<void> {
+    console.log('Clicking footer next button and waiting for window to close');
+    await this.driver.findClickableElement(this.footerNextButton);
+    await this.driver.clickElementAndWaitForWindowToClose(
+      this.footerNextButton,
+    );
+  }
+
+  /**
+   * Check if transfer request message is displayed
+   */
+  async checkTransferRequest(): Promise<void> {
+    console.log('Checking for transfer request message');
+    await this.driver.waitForSelector(this.transferRequestMessage);
+  }
+
+  /**
+   * Check if network text is displayed
+   *
+   * @param networkText - The expected network text to verify
+   */
+  async checkNetwork(networkText: string): Promise<void> {
+    console.log(`Checking for network text: ${networkText}`);
+    await this.driver.waitForSelector(this.networkTextElement(networkText));
+  }
+
+  /**
+   * Open the connection menu
+   */
+  async openConnectionMenu(): Promise<void> {
+    console.log('Opening connection menu');
+    await this.driver.clickElement(this.connectionMenu);
+  }
+
+  /**
+   * Click the connected site popover network button
+   */
+  async clickConnectedSitePopoverNetworkButton(): Promise<void> {
+    console.log('Clicking connected site popover network button');
+    await this.driver.clickElement(this.connectedSitePopoverNetworkButton);
+  }
+
+  /**
+   * Select Ethereum Mainnet from the network options
+   */
+  async selectEthereumMainnet(): Promise<void> {
+    console.log('Selecting Ethereum Mainnet');
+    await this.driver.clickElement(this.ethereumMainnetOption);
+  }
+
+  /**
+   * Switch to Ethereum Mainnet using the connection menu
+   */
+  async switchToEthereumMainnet(): Promise<void> {
+    console.log('Switching to Ethereum Mainnet via connection menu');
+    await this.openConnectionMenu();
+    await this.clickConnectedSitePopoverNetworkButton();
+    await this.selectEthereumMainnet();
+  }
```

Comment:
same, all these methods belong to different metamask pages, not to the test dapp, and most of them already exist.
In the test dapp class, we should only use the selectors that are present in the test dapp, not in another page (ie the MetaMAsk dialog page)